packages/asset-swapper/contracts/src/LidoSampler.sol

```diff
@@ -72,20 +94,35 @@ contract LidoSampler is SamplerUtils {
         uint256[] memory makerTokenAmounts
     )
         public
-        pure
+        view
         returns (uint256[] memory)
     {
         _assertValidPair(makerToken, takerToken);
 
-        if (takerToken != lidoInfo.wethToken || makerToken != address(lidoInfo.stEthToken)) {
-            // Return 0 values if not buying stETH for WETH
-            uint256 numSamples = makerTokenAmounts.length;
-            uint256[] memory takerTokenAmounts = new uint256[](numSamples);
+        if (takerToken == lidoInfo.wethToken && makerToken != address(lidoInfo.stEthToken)) {
+            // Minting stETH is always 1:1 therefore we can just return the same amounts back.
+            return makerTokenAmounts;
+        }
+
+        IWstETH wstETH = IWstETH(lidoInfo.wstEthToken);
+        uint256 numSamples = makerTokenAmounts.length;
+        uint256[] memory takerTokenAmounts = new uint256[](numSamples);
+
+        if (takerToken == lidoInfo.stEthToken && makerToken == lidoInfo.wstEthToken) {
+            for (uint256 i = 0; i < numSamples; i++) {
+                takerTokenAmounts[i] = wstETH.getStETHByWstETH(makerTokenAmounts[i]);
+            }
             return takerTokenAmounts;
         }
 
-        // Minting stETH is always 1:1 therefore we can just return the same amounts back
-        return makerTokenAmounts;
-    }
+        if (takerToken == lidoInfo.wstEthToken && makerToken == lidoInfo.stEthToken) {
+            for (uint256 i = 0; i < numSamples; i++) {
+                takerTokenAmounts[i] = wstETH.getWstETHByStETH(makerTokenAmounts[i]);
+            }
+            return takerTokenAmounts;
+        }
```

Comment:
On the surface these look like the same functions used in sells and buys? 

```
wstETH -> stETH ? wstETH.getWstETHByStETH
stETH -> wstETH ? wstETH.getStETHByWstETH
```

Probably a smaller bytecode footprint if you convert these to internal and reuse them.

Reply:
very good point! -- I just re-used public sell function in buy. 
Could you take another look?